### PR TITLE
feat: connect UI and API

### LINKED ISSUE #1521: github-actions: Integrate GitHub Actions API into frontend view, remove mocks for list of all builds
Implement querying of the github-actions-api to list recent builds of the backstage project itself (hardcoded for now), and showing a bare minimum of metadata about an individual build (such as start time etc)

### PR DESCRIPTION
**Pull request by Nek**
#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [x] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: packages/app/src/apis.ts [hunk 1]
```diff
@@ -82,7 +83,10 @@ export const apis = (config: ConfigApi) => {
     circleCIApiRef,
     new CircleCIApi(`${backendUrl}/proxy/circleci/api`),
   );
-  builder.add(githubActionsApiRef, new GithubActionsClient());
+
+  const octokit = new Octokit();
+  const client = new GithubActionsClient({ api: octokit });
+  builder.add(githubActionsApiRef, client);
```
Comment (shmidt-i) [inline packages/app/src/apis.ts:89]:
```suggestion
  builder.add(githubActionsApiRef, new GithubActionsClient({ api: octokit }));
```

File: plugins/github-actions/package.json [hunk 2]
```diff
@@ -28,6 +28,8 @@
     "@material-ui/core": "^4.9.1",
     "@material-ui/icons": "^4.9.1",
     "@material-ui/lab": "4.0.0-alpha.45",
+    "@octokit/rest": "^18.0.0",
```
Comment (shmidt-i) [inline plugins/github-actions/package.json:31]: If we are passing the octokit client in the constructor, do we still need is in this package as a dependency?

File: plugins/github-actions/src/api/GithubActionsClient.ts [hunk 3]
```diff
@@ -15,117 +15,94 @@
...
-const conclusionToStatus = (conslusion: string): BuildStatus =>
-  statusToBuildStatus[conslusion] ?? BuildStatus.Null;
+import { Octokit } from '@octokit/rest';
+import {
+  ActionsListWorkflowRunsForRepoResponseData,
+  ActionsGetWorkflowResponseData,
+  ActionsGetWorkflowRunResponseData,
+} from '@octokit/types';
+
+// const statusToBuildStatus: { [status: string]: BuildStatus } = {
```
Comment (shmidt-i) [inline plugins/github-actions/src/api/GithubActionsClient.ts:25]: Remove comment?

File: plugins/github-actions/src/api/GithubActionsClient.ts [hunk 4]
```diff
@@ -15,117 +15,94 @@
...
-      transData.status = conclusionToStatus(element.conclusion);
-      transData.message = element.head_commit.message;
-      transData.uri = element.url;
-      endData[index] = transData;
+  }
+  async listWorkflowRuns({
+    owner,
+    repo,
+    pageSize = 100,
+    page = 0,
```
Comment (shmidt-i) [inline plugins/github-actions/src/api/GithubActionsClient.ts:61]: We don't have `token` here anymore, how does it work with auth then?

File: plugins/github-actions/package.json (hunk 2 above)
Comment (Nek) [reply plugins/github-actions/package.json:31]: We do need it as long as we want octokit parameter to be typed.

File: plugins/github-actions/src/api/GithubActionsClient.ts (hunk 3 above)
Comment (Nek) [reply plugins/github-actions/src/api/GithubActionsClient.ts:25]: 👍

File: plugins/github-actions/src/api/GithubActionsClient.ts (hunk 4 above)
Comment (Nek) [reply plugins/github-actions/src/api/GithubActionsClient.ts:61]: I accidentally removed it. Will fix.

**Review by shmidt-i — APPROVED** [review]

**Pull request merged**